python/semantic_kernel/agents/chat_completion/chat_completion_agent.py

```diff
@@ -234,6 +234,31 @@ async def create_channel(
 
         return ChatHistoryChannel(messages=messages, thread=thread)
 
+    # region Declarative Spec
+
+    @override
+    @classmethod
+    async def _from_dict(
+        cls,
+        data: dict,
+        *,
+        kernel: "Kernel | None" = None,
+        **kwargs,
+    ) -> "ChatCompletionAgent":
+        fields = cls._normalize_spec_fields(data, kernel=kernel)
+
+        if "service" in kwargs:
```

Comment:
I need to verify that there aren't other kwargs we should be handling for the ChatCompletionAgent.